src/Http/Routing/src/Builder/EndpointFilterExtensions.cs

```diff
@@ -13,12 +13,15 @@ namespace Microsoft.AspNetCore.Http;
 /// </summary>
 public static class EndpointFilterExtensions
 {
+    internal const string FilterRequiresDynamicCodeWarning = "Filter factories generate dynamic code and aren't compatible with native AOT applications.";
```

Comment:
This isn't true. Filters work fine.

Reply:
If you have filters, we will call into RequestDelegateFactory.Create, which is marked `RequiresDynamicCode`:

https://github.com/dotnet/aspnetcore/blob/9175347822be11e4aa1376e69a4c5f17db731388/src/Http/Routing/src/RouteEndpointDataSource.cs#L228-L235

Reply:
I looked into `RequestDelegateFactory` + filters, and there is logic for falling back to non-dynamic code if AOT:
* https://github.com/dotnet/aspnetcore/blob/589aa11b5c631ce719e0530d66be8324a6d79169/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L546-L551
* https://github.com/dotnet/aspnetcore/blob/589aa11b5c631ce719e0530d66be8324a6d79169/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L611-L619

I think we definitely need AOTed testing of this area. It would be very easy for someone to unknowingly break support.

Reply:
No I think in .NET 8, RequestDelegateFactory will be completely unsafe for AOT. That's what [this work item](https://github.com/dotnet/aspnetcore/issues/45524) is about.

Reply:
Map + RequestDelegate will still go through `RequestDelegateFactory`. I want to test that is ok. And test that it is ok with a filter

Reply:
> Map + RequestDelegate will still go through RequestDelegateFactory (unless that is rewritten).

It will be rewritten.

> I want to test that is ok. And test that Map + RequestDelegate + filter it is ok.

You're testing the current state of the code. Adding filters to an endpoint is not fundamentally unsafe and thus cannot be marked as such.